src/Middleware/Diagnostics/src/DeveloperExceptionPage/DeveloperExceptionPageMiddleware.cs

```diff
@@ -127,13 +128,22 @@ public async Task Invoke(HttpContext context)
         // Assumes the response headers have not been sent.  If they have, still attempt to write to the body.
         private Task DisplayException(ErrorContext errorContext)
         {
-            var compilationException = errorContext.Exception as ICompilationException;
-            if (compilationException != null)
+            var httpContext = errorContext.HttpContext;
+            var acceptHeader = httpContext.Request.Headers[HeaderNames.Accept].ToString();
+
+            // If the client does not ask for HTML just format the exception as plain text
+            if (acceptHeader != null && !acceptHeader.Contains("text/html", StringComparison.OrdinalIgnoreCase))
```

Comment:
yeah but what if the accept header contains `some-cool-text/html` :trollface:  

It's wired to not parse this properly if you're going to look at it.

Reply:
Don’t care. We can complicate it when the bugs come in (which they won’t 😬)

Reply:
Instant debt. Just do it right the first time.

Reply:
Yeah come on please fix this.

Reply:
DONE! Fixed for future browsers we can't predict will ask for html.